README.md

```diff
@@ -31,37 +36,255 @@ npm install @uploadcare/upload-client --save
+
+#### baseURL: string
+
+API base URL. 
+
+Defaults to `https://upload.uploadcare.com`
+
+#### fileName: string
+
+You can specify an original filename. 
+
+Defaults to `original`.
+
+#### doNotStore: boolean
+
+Forces files uploaded with a widget not to be stored. 
+For instance, you might want to turn this on when automatic file storing 
+is enabled in your project, but you do not want to store files uploaded 
+with a particular instance.
+
+#### secureSignature: string
+
+In case you enable signed uploads for your project, you’d need to provide 
+the client with signature and expire. 
+
+The `secureSignature` is an MD5 hex-encoded hash from a concatenation 
+of `API secret key` and `secureExpire`.
+
+#### secureExpire: string
+
+Stands for the Unix time to which the signature is valid, e.g., `1454902434`.
+
+#### integration: string
+
+`X-UC-User-Agent` header value.
+
+Defaults to `UploadcareUploadClient/${version}${publicKey} (JavaScript${integration})`
+
+#### checkForUrlDuplicates: boolean
+
+Runs the duplicate check and provides the immediate-download behavior.
+
+#### saveUrlForRecurrentUploads: boolean
+
+Provides the save/update URL behavior. The parameter can be used 
+if you believe a `sourceUrl` will be used more than once. 
+Using the parameter also updates an existing reference with a new 
+`sourceUrl` content.
+
+#### source: string
+
+Upload source. Each upload method has its own (as example: `local`, `url`, etc.)
+
+#### jsonpCallback: string
+
+Sets the name of your JSONP callback function to create files group from 
+a set of files by using their UUIDs.
+
+#### pollingTimeoutMilliseconds: number
+
+Internally upload client using polling to ensure that file is available on CDN 
+or its uploaded from url.
+
+Defaults to `10000` milliseconds (10 seconds).
+
+#### maxContentLength: number
+
+`maxContentLength` defines the max size of the http response content 
+in bytes allowed. 
+
+Defaults to `52428800` bytes (50 MB).
+
+#### retryThrottledRequestMaxTimes: number
+
+Sets the maximum number of attempts to retry throttled request.
+
+Defaults to `1`.
+
+#### multipartChunkSize: number
+
+This option is only applicable when handling local files.
+Sets the multipart chunk size. 
+
+Defaults to `5242880` bytes (5 MB).
+
+#### multipartMinFileSize: number
+
+This option is only applicable when handling local files. 
+Sets the multipart uploading file size threshold. 
+The value is limited to the range from `10485760` (10 MB) to `104857600` (100 MB). 
+
+Defaults to `26214400` (25 MB).
+
+#### multipartMinLastPartSize: number
+
+This option is only applicable when handling local files.
+Set the minimum size of the last multipart part.
+
+Defaults to `1048576` bytes (1 MB).
+
+#### maxConcurrentRequests: number
+
+Setting that allows you to specify the number of concurrent requests.
+
+Defaults to `4`.
+
 ## Testing 
 
 By default testing environment is local and for this you need to start a mock server for local tests.
```

Comment:
By default, the testing environment is local and requires starting a mock server to run tests.